src/ServerlessPluginBespoken.ts

```diff
@@ -0,0 +1,315 @@
+const mutateProcessEnvironmentVariables = (vars: any) => {
+  for (const key in vars) {
+    process.env[key] = vars[key];
+  }
+  // filter out ridiculous number of npm_ env variables
+  for (const key in process.env) {
+    if (key.startsWith("npm_")) {
+      delete process.env[key];
+    }
+  }
+};
+
+/**
+ * Defines command:
+ * - `serverless proxy`
+ *    Initiates connection to public bespoken proxy so requests to public url are forwarded to local machine and starts 
+ *    a local server that responds to requests by dispatching the url path to the appropriate lambda function.
+ * 
+ * - `serverless deploy --enablePassThru`
+ *   When --enablePassThru is passed on command line, the plugin will intercept the serverless package lifecycle and replace the deployed bundle with passthru lambda handlers
+ *   that forward requests to the bespoken proxy server (and subsequently back to developer machine)
+ */
+export class ServerlessPluginBespoken {
+  private serverless: any;
+  private options: any;
+  private proxy: BSTProxy;
+  private originalServicePath: string;
+  private passThruServicePath: string;
+
+  private get environmentVariablesFromServerlessConfig(): object {
+    const env = this.serverless.service.provider.environment;
+    if (env) {
+      this.serverless.cli.log(
+        "Configuring environment variables from serverless config"
+      );
+    } else {
+      this.serverless.cli.log(
+        "No Environment variables found in serverless config"
+      );
+    }
+    return env || {};
+  }
+
+  private get handlers(): string[] {
+    const handlers = this.serverless.service.functions || {};
+
+    return Object.entries(handlers).map(([_, lambdaDefinition]) => {
+      return lambdaDefinition.handler;
+    });
+  }
+
+  public commands = {
+    proxy: {
+      usage: "Plugin to call bespoken bst lambda service",
+      lifecycleEvents: ["start"],
+      options: {
+        function: {
+          usage:
+            "Specify the function in your config that you want to use" +
+            '(e.g. "--function myFunction" or "-f myFunction")',
+          required: false,
+          shortcut: "f"
+        }
+      }
+    }
+  };
+
+  public get hooks() {
+    return {
+      "proxy:start": this.proxyStart,
+      "before:package:createDeploymentArtifacts": this.deployPassThru,
+      "after:package:createDeploymentArtifacts": this.injectPassThruModules
+    };
+  }
+
+  constructor(serverless: any, options: any) {
+    this.serverless = serverless;
+    this.options = options;
+  }
+
+  proxyStart = async () => {
+    // parse the bespoken config
+    await Global.initializeCLI();
+    // enable verbose logging
+    LoggingHelper.setVerbose(true);
+
+    // ensure environment variables from serverless config are set
+    mutateProcessEnvironmentVariables(
+      this.environmentVariablesFromServerlessConfig
+    );
+
+    // create and start the lambda proxy
+    this.proxy = BSTProxy.lambda();
+    this.proxy.activateSecurity();
```

Comment:
Security could be a passthrough requirement but for a simple "sls proxy" without parameters it should be off to mimic default bst proxy lambda index.js functionality

Reply:
👍

Reply:
Oops I misread this request initially ... was thinking you wanted activateSecurity off when --function is present and on otherwise.  

I just added --secure option to control activateSecurity usage which is off by default.